### DISCUSSION TITLE
A generic value for APPWRITE_FUNCTION_ENDPOINT

### DISCUSSION THREAD

**Discussion by merabtenei** [appwrite/appwrite#4630] [Ideas]
I was wondering why setting APPWRITE_FUNCTION_ENDPOINT to http://traefik/v1 was not working. I'm a bit familiar with docker compose and I know that services can access other services in docker by specifying the service name as a host. Why is this not working with Appwrite functions.
And finally isn't there any way to make a generic solution, a value that would work for any situation as long as functions are executed in the same server as Appwrite ?

**Comment by stnguyen90** [comment]
The reason why it doesn't work is because the runtime container is not added to the same docker network as the traefik container. The downside about using that as the default endpoint is you'd have to use a self signed certificate for HTTPS. There's probably a workaround, but it can get complicated with various edge cases.

**Reply by merabtenei** [reply]
Do we really need https when communicating between containers on the same machine ?

**Reply by stnguyen90** [reply]
Maybe not, but Appwrite could be configured to automatically redirect to https which could cause problems with some requests.

**Comment by zlmr** [comment]
I would also like to have it at some point - it would probably make function execution even faster.
Is it related to this issue https://github.com/appwrite/appwrite/issues/2694?

**Comment by merabtenei** [comment]
@zlmr It's not really about performance, I don't know if this can make functions faster but it's more about migrating from a server to another. If we have many functions we need to redefine all those ENV variables as the CLI doesn't export those. Making migrations from a test server to a production server hard to automate.

**Reply by stnguyen90** [reply]
FYI, the Appwrite CLI van deploy variables

**Reply by merabtenei** [reply]
I didn't find anything referencing to this in the doc. I didn't find any conclusion in the following discussion too : https://github.com/appwrite/appwrite/discussions/2985#discussioncomment-2420245

**Reply by stnguyen90** [reply]
@merabtenei, ya...we need to document the appwrite.json a little better. Deploying function variables was kind of a recent feature. You can add functions by adding `variables` to the function like:

```json
        {
            "$id": "my-function",
            "name": "my-function",
            "runtime": "dart-2.17",
            "path": "functions/my-function",
            "entrypoint": "lib/main.dart",
            "ignore": [".packages", ".dart_tool"],
            "execute": [],
            "variables": {
                "APPWRITE_FUNCTION_ENDPOINT": "https://[HOSTNAME]/v1",
                "APPWRITE_FUNCTION_API_KEY": "[API KEY]"
            },
            "events": ["databases.*.create"],
            "schedule": "",
            "timeout": 15
        }
```

When running `appwrite deploy function` you'll be warned that variables on your server will be wiped and replaced with what's defined in your appwrite.json.

**Reply by merabtenei** [reply]
@stnguyen90 I had the chance to test that today, and it seems it's not working add variables attribute in the .json file, deployed functions on a new server, after accessing the appwrite console I see nothing in Settings -> Update Function Variables of any function deployed.
I use appwrite version 1.1.0 and appwrite-cli version 1.1.1, Is it supposed to work or is this not yet included on the last release ?
```
"variables": {
                "APPWRITE_FUNCTION_ENDPOINT": "https://[HOSTNAME]/v1",
                "APPWRITE_FUNCTION_API_KEY": "[API KEY]"
            },
```

**Reply by stnguyen90** [reply]
@merabtenei it's been working even in 1.0.x. There might be a bug where the variables aren't deployed the first time you deploy the function, but it should work after that. Did you get a warning about the variables on the server will be wiped?

**Reply by merabtenei** [reply]
@stnguyen90 Correct, I didn't get the variables at the first time. but after a second deploy or some time I got the variables showing. Don't know exactly how it works but it takes some time to see them displayed on the console.
